src/lib.rs

```diff
@@ -131,13 +131,9 @@ impl<T: AsMut<[u8]>> LogBuffer<T> {
             byte & 0b11111000 == 0b11110000
         }
 
-        let buffer = self.buffer.as_mut();
-        for i in 0..buffer.len() {
-            if is_utf8_leader(buffer[i]) {
-                return core::str::from_utf8(&buffer[i..]).unwrap()
-            }
-        }
-        return ""
+        let buffer = &*self.buffer.as_mut();
+        buffer.iter().cloned().position(is_utf8_leader).map_or("", |i|
```

Comment:
I'm confused. Why did you add `&*`? What's the purpose of `.cloned()` here?

I don't really see how this transformation makes the code more clear...